packages/studio-base/src/components/TimeBasedChart/downsample.ts

```diff
@@ -23,7 +23,8 @@ export const MINIMUM_PIXEL_DISTANCE = 3;
 
 type IntervalItem = { xPixel: number; yPixel: number; label: string | undefined; index: number };
 
-export type State = {
+// Contains the state of an ongoing downsample operation.
```

Comment:
```suggestion
/** Contains the state of an ongoing downsample operation. */
```

One thing you've lost with your new separate functions is actually documenting the downsample approach. Notice how the original function told you what it did - this is no longer true with your new functions and if we deleted the original there's no user-facing documentation on the intent of our downsampling.